app/controllers/schedule_rules_controller.rb

```diff
@@ -16,83 +16,89 @@ def initialize
+      start_hour: 9,
+      start_min: 0,
+      end_hour: 17,
+      end_min: 0,
+    )
   end
 
   # POST /schedule_rules
   def create
-    @schedule_rule  = @instrument.schedule_rules.new(params[:schedule_rule])
+    @schedule_rule = @product.schedule_rules.new(schedule_rule_params)
 
-    respond_to do |format|
-      if @schedule_rule.save
-        flash[:notice] = "Schedule Rule was successfully created."
-        format.html { redirect_to(facility_instrument_schedule_rules_path(current_facility, @instrument)) }
-      else
-        format.html { render action: "new" }
-      end
+    if @schedule_rule.save
+      flash[:notice] = text("create")
+      redirect_to action: :index
+    else
+      render action: "new"
     end
   end
 
+  # GET /schedule_rules/1/edit
+  def edit
+  end
+
   # PUT /schedule_rules/1
   def update
-    @schedule_rule = ScheduleRule.find(params[:id])
-    # TODO: 404 protection for non inst, facil rules
-
-    # if there are no boxes checked, remove them all
+    # if there are no boxes checked, the empty array will mark the
+    # existing ones for destruction
     params[:schedule_rule][:product_access_group_ids] ||= []
-    respond_to do |format|
-      if @schedule_rule.update_attributes(params[:schedule_rule])
-        flash[:notice] = "Schedule Rule was successfully updated."
-        format.html { redirect_to(facility_instrument_schedule_rules_path(current_facility, @instrument)) }
-      else
-        format.html { render action: "edit" }
-      end
+
+    if @schedule_rule.update_attributes(schedule_rule_params)
+      flash[:notice] = text("update")
+      redirect_to action: :index
+    else
+      render action: "edit"
     end
   end
 
   # DELETE /schedule_rules/1
   def destroy
-    @schedule_rule = ScheduleRule.find(params[:id])
-    # TODO: 404 on inappropriate schedule rules
     @schedule_rule.destroy
 
-    respond_to do |format|
-      format.html { redirect_to(facility_instrument_schedule_rules_path(current_facility, @instrument)) }
-    end
+    flash[:notice] = text("destroy")
+    redirect_to action: :index
   end
 
   private
 
-  def init_instrument
-    @instrument = current_facility.instruments.find_by_url_name!(params[:instrument_id])
+  def schedule_rule_params
+    params.require(:schedule_rule).permit(:discount_percent, :start_hour, :start_min, :end_hour, :end_min,
+                                          :on_sun, :on_mon, :on_tue, :on_wed, :on_thu, :on_fri, :on_sat,
+                                          product_access_group_ids: [])
+  end
+
+  def init_product
+    @product = current_facility.products.find_by!(url_name: product_id)
+  end
+
+  def init_schedule_rule
+    @schedule_rule = @product.schedule_rules.find(params[:id])
+  end
+
+  def product_id
+    params[product_key]
   end
 
-  def init_current_product
-    # required for tabnav_product
-    @product = @instrument
+  def product_key
+    params.except(:facility_id).keys.find { |k| k.end_with?("_id") }
```

Comment:
maybe check explicitly for one of [:instrument_id, :item_id, :service_id, :bundle_id] so that foo_id can't spoof this?